File: wiki-backend/src/main/java/com/internal/team/wiki/dto/doc/response/DocsResponse.java
package com.internal.team.wiki.dto.doc.response;

import com.internal.team.wiki.domain.doc.DocEntity;
import lombok.Getter;

import java.util.List;
import java.util.stream.Collectors;

@Getter
public class DocsResponse {

    private final List<DocResponse> responses;

    public DocsResponse(final List<DocResponse> responses) {
        this.responses = responses;
    }

    public static DocsResponse of(final List<DocEntity> docEntities) {
        List<DocResponse> responses = docEntities.stream()
                .map(DocResponse::from)
                .collect(Collectors.toList());
        return new DocsResponse(responses);
    }
}

File: wiki-backend/src/main/java/com/internal/team/wiki/doc/dto/DocsResponse.kt
package com.internal.team.wiki.doc.dto

import com.internal.team.wiki.doc.domain.DocEntity

data class DocsResponse(
    val responses: List<DocResponse>
) {
    companion object {
        @JvmStatic
        fun of(docEntities: List<DocEntity>): DocsResponse {
            val responses = docEntities.map {
                DocResponse.from(it)
            }
            return DocsResponse(responses)
        }
    }
}